contracts/AaveMiddleware.sol

```diff
@@ -0,0 +1,180 @@
+    address wethGatewayAddress = 0xcc9a0B7c43DC2a5F023Bb9b738E45B0Ef6B06E04; //Mainnet
+    address _LendingPoolAddress = getLendingPoolAddress();
+    IWETHGateway wethGateway = IWETHGateway(wethGatewayAddress);
+    wethGateway.depositETH{ value: msg.value }(
+      _LendingPoolAddress,
+      address(this),
+      0
+    );
+  }
+
+  function withdrawEth(uint256 _amount) external {
+    address wethGatewayAddress = 0xcc9a0B7c43DC2a5F023Bb9b738E45B0Ef6B06E04; //Mainnet
+    address _LendingPoolAddress = getLendingPoolAddress();
+    IWETHGateway wethGateway = IWETHGateway(wethGatewayAddress);
+    address aWethAddress = 0x030bA81f1c18d280636F32af80b9AAd02Cf0854e; //AWeth contract on Mainnet
+    IERC20 erc20Token = IERC20(aWethAddress);
+    erc20Token.safeApprove(wethGatewayAddress, _amount);
+    wethGateway.withdrawETH(_LendingPoolAddress, _amount, msg.sender);
+  }
+
+  function borrowEth(uint256 _amount) external payable {
+    address wethAddress = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2; //Weth on Ethereum
+    address _LendingPoolAddress = getLendingPoolAddress();
+    ILendingPool LendingPool = ILendingPool(_LendingPoolAddress);
+    LendingPool.borrow(wethAddress, _amount, 1, 0, address(this)); //Amount borrowed. Will be received by the contract
+
+    Iweth weth = Iweth(wethAddress);
+    weth.withdraw(_amount);
+    bool sent = payable(msg.sender).send(address(this).balance);
+    console.log(sent);
+  }
+
+  function repayEth() external payable {
+    uint256 _amount = msg.value;
+    address wethAddress = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2; //Weth on Ethereum
+    Iweth weth = Iweth(wethAddress);
+    weth.deposit{ value: msg.value }();
+
+    IERC20 erc20Token = IERC20(wethAddress);
+    address _LendingPoolAddress = getLendingPoolAddress();
+    ILendingPool LendingPool = ILendingPool(_LendingPoolAddress);
+    erc20Token.safeApprove(_LendingPoolAddress, _amount);
+    LendingPool.repay(wethAddress, _amount, 1, address(this)); //Amount borrowed
+  }
+
+  function leverage(uint256 _amount) external payable {
+    address _LendingPoolAddress = getLendingPoolAddress();
+    ILendingPool LendingPool = ILendingPool(_LendingPoolAddress);
+    (
+      uint256 a1,
+      uint256 a2,
+      uint256 a3,
+      uint256 a4,
+      uint256 a5,
+      uint256 a6
+    ) = LendingPool.getUserAccountData(address(this));
+    console.log("totalCollateralETH         ", a1);
+    console.log("totalDebtETH               ", a2);
+    console.log("availableBorrowsETH        ", a3);
+    console.log("currentLiquidationThreshold", a4);
+    console.log("ltv                        ", a5);
+    console.log("healthFactor               ", a6);
+    console.log("Deposit");
+    console.log("Amount requested to Borrow", _amount);
+
+    require(_amount <= a3, "Requested Amount is more than borrow allowance");
+
+    console.log("Borrowing..");
+    address wethAddress = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2; //Weth on Ethereum
+
```

Comment:
Let's add one condition here that the user should have supplied before trying to leverage?

Reply:
Wouldn't that be taken care of by the require line? Like if the user hasn't supplied anything, then the maxBorrowEth will be zero and hence requested amount will be more than what is max available and it will be denied. 
Should I do a seperate check for this?

Reply:
If yes, What kind of check? Should I just put a require condition and check that collateral despoited on aave should not be zero?

Reply:
So, we do a quick check by creating a mapping of `address->uint256`, which will be user and amt that the user-supplied. So, if the user has supplied some amt, then we have the check condition in hand.

Reply:
Okay Done!